command_line/slice_sequence.py

```diff
@@ -158,33 +160,42 @@ def run(self, args=None):
 
         # check if slicing into blocks
         if params.block_size is not None:
-            if slice_exps:
-                if len(experiments) > 1:
-                    raise Sorry(
-                        "For slicing into blocks please provide a single " "scan only"
-                    )
-                scan = experiments[0].scan
+            if not slice_exps:
+                raise Sorry(
+                    "For slicing into blocks, an experiment file must be provided"
+                )
+
+            if len(experiments) > 1:
+                raise Sorry(
+                    "For slicing into blocks please provide a single " "scan only"
+                )
+            scan = experiments[0].scan
 
             # Having extracted the scan, calculate the blocks
             params.image_range = calculate_block_ranges(scan, params.block_size)
 
             # Do the slicing then recombine
-            if slice_exps:
-                sliced = [
-                    slice_experiments(experiments, [sr])[0] for sr in params.image_range
-                ]
-                sliced_experiments = ExperimentList()
-                for exp in sliced:
-                    sliced_experiments.append(exp)
+            sliced = [
+                slice_experiments(experiments, [sr])[0] for sr in params.image_range
+            ]
+            sliced_experiments = ExperimentList()
+            generate_experiment_identifiers(sliced)
+            for exp in sliced:
+                sliced_experiments.append(exp)
 
             # slice reflections if present
             if slice_refs:
                 sliced = [
                     slice_reflections(reflections, [sr]) for sr in params.image_range
                 ]
-                sliced_reflections = sliced[0]
-                for i, rt in enumerate(sliced[1:]):
-                    rt["id"] += i + 1  # set id
+                sliced_reflections = flex.reflection_table()
+                identifiers = sliced_experiments.identifiers()
```

Comment:
A comment around this block along the lines of "resetting experiment identifiers" might be useful for future readers